### PR TITLE
[MDB IGNORE] Adds summon beacons for command pets, secmed locker, and commandrobes

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
This PR moves the command pets, secmed locker, and command outfitting stations into their beacon form.
They can only be used in specific areas.
Security medics now spawn in the same locations as security officers
They are found in the following places:

- Markus: QM locker
- E-N: RD locker
- Bumbles: HoP locker
- Poppy: CE locker
- Commanddrobe: Cap locker
- Secmed locker: secmed backpack

image
image

## How This Contributes To The Skyrat Roleplay Experience
Less stuff to upkeep in our maps, which is good.

## Changelog

:cl:
refactor: Command pets, the command outfitting station, and secmed locker are now called in through deployment beacons, found in their respective lockers (pets and outfitting in head of staff lockers) or on their person (secmed's locker).
/:cl:

### CODE REVIEW

**Comment by Mryourface13** [comment]
YOOOO SECMED LOCKER IN BACKPACK

**Review by Jolly-66 — CHANGES_REQUESTED** [review]
Edit the templates to either remove the entirety of secmed, or just the locker spawner.

File: modular_skyrat/modules/modular_items/code/summon_beacon.dm [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	if(!display_names[choice])
+		return
+	if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
+		return
+	selected_atom = display_names[choice]
+
+/obj/item/summon_beacon/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!selected_atom)
+		balloon_alert(user, "no atom selected!")
```
Comment (tf-4) [inline modular_skyrat/modules/modular_items/code/summon_beacon.dm]:
```suggestion
		balloon_alert(user, "no choice selected!")
```
The average player won't know what an atom is

File: modular_skyrat/modules/modular_items/code/summon_beacon.dm [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+		return
+	var/turf/target_turf = get_turf(target)
+	var/area/target_area = get_area(target)
+	if(!target_turf || !target_area || !is_type_in_list(target_area, allowed_areas))
+		balloon_alert(user, "can't call here!")
+		return
+
+	var/confirmed = tgui_alert(user, "Are you sure you want to call it here?", "Confirmation", list("Yes", "No"))
```
Comment (tf-4) [inline modular_skyrat/modules/modular_items/code/summon_beacon.dm]:
```suggestion
	var/confirmed = tgui_alert(user, "Are you sure you want to call [selected_atom.name] here?", "Confirmation", list("Yes", "No"))
```
Something I was meaning to do with the miner beacons anyway

File: modular_skyrat/modules/modular_items/code/summon_beacon.dm [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+/obj/item/summon_beacon/examine()
+	. = ..()
+	. += span_warning("Caution: Only works in [area_string].")
+	. += span_notice("Currently selected: [selected_atom ? selected_atom.name : "None"].")
+
+/obj/item/summon_beacon/attack_self(mob/user)
+	if(!can_use_beacon(user))
+		return
+	show_options(user)
```
Comment (tf-4) [inline modular_skyrat/modules/modular_items/code/summon_beacon.dm:42]:
```suggestion
	if(length(selectable_atoms) == 1)
		selected_atom = selectable_atoms[1]
		return
	show_options(user)
```
No sense having the user choose from a list if they don't have any actual options

File: modular_skyrat/modules/modular_items/code/summon_beacon.dm [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+		/obj/machinery/atmospherics/miner/n2o,
+		/obj/machinery/atmospherics/miner/nitrogen,
+		/obj/machinery/atmospherics/miner/oxygen,
+		/obj/machinery/atmospherics/miner/plasma,
+	)
+
+	area_string = "atmospherics"
+	supply_pod_stay = TRUE
+
+/obj/item/summon_beacon/commanddrobe
```
Comment (tf-4) [inline modular_skyrat/modules/modular_items/code/summon_beacon.dm]:
```suggestion
/obj/item/summon_beacon/command_drobe
```

**Review by tf-4 — COMMENTED** [review]
good pr

File: code/game/objects/structures/crates_lockers/closets/secure/security.dm [hunk 5]
```diff
@@ -20,6 +20,7 @@
 	new /obj/item/storage/box/gunset/pdh_captain(src) // SKYRAT EDIT ADDITION
 	new /obj/item/door_remote/captain(src)
 	new /obj/item/storage/photo_album/captain(src)
+	new /obj/item/summon_beacon/commanddrobe(src) // SKYRAT EDIT ADDITION
```
Comment (Zonespace27) [inline code/game/objects/structures/crates_lockers/closets/secure/security.dm]:
```suggestion
	new /obj/item/summon_beacon/command_drobe(src) // SKYRAT EDIT ADDITION
```

**Comment by desertmote** [comment]
e-n, poppy and bumbles were designed as pet of non-heads though :/

e-n is robotics,  poppy is engineering and bumbles is botany. id recommend putting these beacons instead in their departments break area or something thats still keeping the point of this PR

**Comment by desertmote** [comment]
id also ask to leave them plainly in blueshift? this is a compromise which would be needlessly applied to our modular map

**Comment by desertmote** [comment]
why not just code a modular script that places the pets into the correct zone on round initialize?

**Review by Jolly-66 — APPROVED** [review]
Maps LGTM. I'll organize the template map at a later date.

Also double check what I sent you on Discord.

**Review by Gandalf2k15 — CHANGES_REQUESTED** [review]
There is already an advanced choice beacon system which you should expand upon instead of making a new one, it's advanced_choice_beacon I think

**Comment by Zonespace27** [comment]
> e-n, poppy and bumbles were designed as pet of non-heads though :/
> 
> e-n is robotics, poppy is engineering and bumbles is botany. id recommend putting these beacons instead in their departments break area or something thats still keeping the point of this PR

putting the beacons in the maps themselves defeats the point of the PR, to lighten the load on map resets

**Comment by Zonespace27** [comment]
> id also ask to leave them plainly in blueshift? this is a compromise which would be needlessly applied to our modular map

consistency, we don't need two of each pet, two secmed lockers, and two commandrobes

**Comment by Zonespace27** [comment]
> There is already an advanced choice beacon system which you should expand upon instead of making a new one, it's advanced_choice_beacon I think

I looked at the advanced choice beacon and decided to remove it, moving its functionality to the new beacon type

**Comment by Gandalf2k15** [comment]
The advanced choice beacon used a radial menu, incorporate that eeee

**Comment by tf-4** [comment]
> The advanced choice beacon used a radial menu, incorporate that eeee

That won't really work for stuff like gas miners which all look the same

**Comment by GoldenAlpharex** [comment]
> > The advanced choice beacon used a radial menu, incorporate that eeee
> 
> That won't really work for stuff like gas miners which all look the same

Doesn't it have a tooltip when you hover over one of the options in the radial menu, though?

**Comment by Gandalf2k15** [comment]
> > > The advanced choice beacon used a radial menu, incorporate that eeee
> > 
> > 
> > That won't really work for stuff like gas miners which all look the same
> 
> Doesn't it have a tooltip when you hover over one of the options in the radial menu, though?

It has a tooltip

**Comment by Zonespace27** [comment]
> There is already an advanced choice beacon system which you should expand upon instead of making a new one, it's advanced_choice_beacon I think

Added radial support

File: modular_skyrat/modules/modular_items/code/summon_beacon.dm [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+
+	/// Descriptor of what area it should work in
+	var/area_string = "any"
+
+	/// If the supply pod should stay or not
+	var/supply_pod_stay = FALSE
+
+/obj/item/summon_beacon/examine()
+	. = ..()
+	. += span_warning("Caution: Only works in [area_string].")
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/modular_items/code/summon_beacon.dm:33]: cant this just be initial(allowed_area or something

Comment (Zonespace27) [reply modular_skyrat/modules/modular_items/code/summon_beacon.dm:33]: allowed_area is a list

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**